composer/src/main/kotlin/com/gojuno/composer/html/HtmlReport.kt

```diff
@@ -40,3 +88,31 @@ fun writeHtmlReport(gson: Gson, suites: List<Suite>, outputDir: File): Completab
  * See https://youtrack.jetbrains.com/issue/KT-14056
  */
 fun File.relativePathTo(base: File): String = absoluteFile.toRelativeString(base.absoluteFile)
+
+fun inputStreamFromResources(path: String): InputStream = Suite::class.java.classLoader.getResourceAsStream(path)
+
+fun generateLogcatHtml(logcatOutput: File): String = when (logcatOutput.exists()) {
+    false -> ""
+    true -> logcatOutput
+            .readLines()
+            .map { line -> """<div class="log__${cssClassForLogcatLine(line)}">${StringEscapeUtils.escapeXml11(line)}</div>""" }
+            .fold(StringBuilder("""<div class="content"><div class="card log">""")) { stringBuilder, line ->
+                stringBuilder.appendln(line)
+            }
+            .let { it.appendln("""</div></div>""") }
+            .let { it.toString() }
+}
+
+fun cssClassForLogcatLine(logcatLine: String): String {
+    // Logcat line example: `06-07 16:55:14.490  2100  2100 I MicroDetectionWorker: #onError(false)`
```

Comment:
The logcat line in comment actually contain date and id-s in the beginning, not loglevel.